sdk/core-rest/core-client/src/getClient.ts

```diff
@@ -0,0 +1,127 @@
+ * Creates a client with a default pipeline
+ * @param baseUrl Base endpoint for the client
+ * @param credentials Credentials to authenticate the requests
+ * @param options Client options
+ */
+export function getClient(
+  baseUrl: string,
+  credentials?: TokenCredential | KeyCredential,
+  options?: PipelineOptions
+): Client;
+export function getClient(
+  baseUrl: string,
+  credentialsOrPipelineOptions?: (TokenCredential | KeyCredential) | PipelineOptions,
+  opts: PipelineOptions = {}
+): Client {
+  let credentials: TokenCredential | KeyCredential | undefined;
+  let options = opts;
+
+  if (credentialsOrPipelineOptions) {
+    if (isCredential(credentialsOrPipelineOptions)) {
+      credentials = credentialsOrPipelineOptions;
+      options = opts;
+    } else {
+      options = credentialsOrPipelineOptions || {};
+    }
+  }
+
+  const pipeline = createDefaultPipeline(baseUrl, credentials, options);
+  const client = (path: string, ...args: Array<any>) => {
+    return {
+      get: (options: RequestParameters = {}): Promise<HttpResponse> => {
+        const url = buildRequestUrl(baseUrl, path, args, options);
+        return sendRequest("GET", url, pipeline, options);
+      },
+      post: (options: RequestParameters = {}): Promise<HttpResponse> => {
+        const url = buildRequestUrl(baseUrl, path, args, options);
+        return sendRequest("POST", url, pipeline, options);
+      },
+      put: (options: RequestParameters = {}): Promise<HttpResponse> => {
+        const url = buildRequestUrl(baseUrl, path, args, options);
+        return sendRequest("PUT", url, pipeline, options);
+      },
+      patch: (options: RequestParameters = {}): Promise<HttpResponse> => {
+        const url = buildRequestUrl(baseUrl, path, args, options);
+        return sendRequest("PATCH", url, pipeline, options);
+      },
+      delete: (options: RequestParameters = {}): Promise<HttpResponse> => {
+        const url = buildRequestUrl(baseUrl, path, args, options);
+        return sendRequest("DELETE", url, pipeline, options);
+      },
+      head: (options: RequestParameters = {}): Promise<HttpResponse> => {
+        const url = buildRequestUrl(baseUrl, path, args, options);
+        return sendRequest("HEAD", url, pipeline, options);
+      },
+      options: (options: RequestParameters = {}): Promise<HttpResponse> => {
+        const url = buildRequestUrl(baseUrl, path, args, options);
+        return sendRequest("OPTIONS", url, pipeline, options);
+      },
+      trace: (options: RequestParameters = {}): Promise<HttpResponse> => {
+        const url = buildRequestUrl(baseUrl, path, args, options);
+        return sendRequest("TRACE", url, pipeline, options);
+      },
+    };
+  };
+
+  return {
+    path: client,
+    pathUnchecked: client,
+  };
+}
+
+function isCredential(
+  param: (TokenCredential | KeyCredential) | PipelineOptions
+): param is TokenCredential | KeyCredential {
+  if ((param as any).key || isTokenCredential(param)) {
```

Comment:
This will return false if `param.key` is an empty string. An empty key is definitely an error, but maybe one that we want to handle differently than by assuming `param` is a PipelineOptions.

```suggestion
  if (((param as KeyCredential).key === undefined) || isTokenCredential(param)) {
```